docs/overview.rst

```diff
@@ -1,380 +1,306 @@
-    # process, as it recycles underlying TCP/IP network connections between
-    # your process and Ethereum node
-    >>> w3 = Web3(HTTPProvider('http://localhost:8545'))
+   # IPCProvider:
+   >>> w3 = Web3(Web3.IPCProvider('./path/to/geth.ipc'))
 
-    # or for an IPC based connection
-    >>> w3 = Web3(IPCProvider())
+   # HTTPProvider:
+   >>> w3 = Web3(Web3.HTTPProvider('http://127.0.0.1:8545'))
 
-    # or for Websocket based connection
-    >>> w3 = Web3(WebsocketProvider('ws://127.0.0.1:8546'))
+   # WebsocketProvider:
+   >>> w3 = Web3(Web3.WebsocketProvider('ws://127.0.0.1:8546'))
 
+   >>> w3.isConnected()
+   True
 
-Base API
---------
-
-The ``Web3`` class exposes the following convenient APIs.
+For more information, (e.g., connecting to remote nodes, provider auto-detection, 
+using a test provider) see the :ref:`Providers <providers>` documentation.
 
 
-.. _overview_type_conversions:
+Middleware
+----------
 
-Type Conversions
-~~~~~~~~~~~~~~~~
-
-.. py:method:: Web3.toHex(primitive=None, hexstr=None, text=None)
-
-    Takes a variety of inputs and returns it in its hexadecimal representation.
-    It follows the rules for converting to hex in the
-    `JSON-RPC spec`_
-
-    .. code-block:: python
-
-        >>> Web3.toHex(0)
-        '0x0'
-        >>> Web3.toHex(1)
-        '0x1'
-        >>> Web3.toHex(0x0)
-        '0x0'
-        >>> Web3.toHex(0x000F)
-        '0xf'
-        >>> Web3.toHex(b'')
-        '0x'
-        >>> Web3.toHex(b'\x00\x0F')
-        '0x000f'
-        >>> Web3.toHex(False)
-        '0x0'
-        >>> Web3.toHex(True)
-        '0x1'
-        >>> Web3.toHex(hexstr='0x000F')
-        '0x000f'
-        >>> Web3.toHex(hexstr='000F')
-        '0x000f'
-        >>> Web3.toHex(text='')
-        '0x'
-        >>> Web3.toHex(text='cowmö')
-        '0x636f776dc3b6'
+Your Web3.py instance may be further configured via middleware.
 
-.. _JSON-RPC spec: https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding
+Web3.py middleware is described using an onion metaphor, where each layer of
+middleware may affect both the incoming request and outgoing response from your
+provider. The documentation includes a :ref:`visualization <Modifying_Middleware>`
+of this idea.
 
-.. py:method:: Web3.toText(primitive=None, hexstr=None, text=None)
+Several middleware are :ref:`included by default <default_middleware>`. You may add to
+(:meth:`add <Web3.middleware_onion.add>`, :meth:`inject <Web3.middleware_onion.inject>`,
+:meth:`replace <Web3.middleware_onion.replace>`) or disable
+(:meth:`remove <Web3.middleware_onion.remove>`,
+:meth:`clear <Web3.middleware_onion.clear>`) any of these middleware.
 
-    Takes a variety of inputs and returns its string equivalent.
-    Text gets decoded as UTF-8.
 
+Your Keys
+~~~~~~~~~
 
-    .. code-block:: python
+Private keys are required to approve any transaction made on your behalf. The type of
+provider you use will determine how you interact with those keys and will affect the
```

Comment:
The provider type doesn't actually dictate this.  Maybe a more accurate statement would be:

> The manner in which your key is secured will determine how you create and send transactions in Web3.py.